cordova-lib/src/cordova/platform.js

```diff
@@ -76,97 +77,94 @@ function add(hooksRunner, projectRoot, targets, opts) {
     }
 
     return hooksRunner.fire('before_platform_add', opts)
-    .then(function() {
-        return promiseutil.Q_chainmap(targets, function(target) {
-            // For each platform, download it and call its "create" script.
-            var parts = target.split('@');
-            var platform = parts[0];
-            var version = parts[1];
-
-            return Q.when().then(function() {
-                if (!(platform in platforms)) {
-                    return getPlatformDetailsFromDir(target);
-                } else {
-                    if (!version) {
-                        events.emit('verbose', 'No version supplied. Retrieving version from config.xml...');
-                    }
-                    version = version || getVersionFromConfigFile(platform, cfg);
-                    var tgt = version ? (platform + '@' + version) : platform;
-                    return isDirectory(version) ? getPlatformDetailsFromDir(version) : downloadPlatform(projectRoot, tgt, opts);
-                }
-            }).then(function(platDetails) {
-                var template = config_json && config_json.lib && config_json.lib[platform] && config_json.lib[platform].template || null;
-                return call_into_create(platDetails.platform, projectRoot, cfg, platDetails.libDir, template, opts);
-            });
-        });
-    })
-    .then(function() {
-        return hooksRunner.fire('after_platform_add', opts);
-    });
+	.then(function () {
```

Comment:
please don't insert tabs. and please try to avoid rewriting large chunks of files when adding features.